lib/solveEquation/stepThrough.js

```diff
@@ -50,9 +51,23 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   const MAX_STEP_COUNT = 20;
   let iters = 0;
 
+  // Checks if there are roots in the original equation before we
```

Comment:
@evykassirer does that make sense to you? If that's the case, @aliang8 can you make it a helper function?